src/js/engagement_view/src/components/engagementView/sidebar/utils/toggleLensTable.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useEffect, useState } from "react";
+
+import Button from "@material-ui/core/Button";
+import KeyboardArrowDownOutlinedIcon from '@material-ui/icons/KeyboardArrowDownOutlined';
+import KeyboardArrowUpOutlinedIcon from '@material-ui/icons/KeyboardArrowUpOutlined';
+
+import Divider from "@material-ui/core/Divider";
+import Backdrop from "@material-ui/core/Backdrop";
+import CircularProgress from "@material-ui/core/CircularProgress";
+
+import { lensTable } from "./lensTable";
+import { getLenses } from "services/graphQLRequests/getLenses";
+
+import { ToggleLensTableProps, ToggleLensTableState } from "types/CustomTypes";
+
+import { useStyles } from "../styles";
+
+const defaultToggleLensTableState = (): ToggleLensTableState => {
+	return {
+		toggled: true,
+		lenses: [],
+		first: 100, // first is the page size
+		offset: 0, // by default, start from page 0
+	};
+};
+
+export function ToggleLensTable({ setLens }: ToggleLensTableProps) {
+	const classes = useStyles();
+
+	const [lensRetrievedState, setLensRetrievedState] = useState(null);
+	const [toggleTableState, setToggleTableState] = useState(
+		defaultToggleLensTableState()
+	);
+	const [pageState, setPageState] = useState(0);
+	const [rowsPerPageState, setRowsPerPageState] = useState(10);
```

Comment:
Is it normal to have multiple `useState` calls in a react component? I would expect instead to have a single `useState` with an object that maintains that state vs multiple use states where each one has its own value to manage.

Reply:
Yes, this is normal and advised by the authors of `react-force-graph`. See: https://github.com/vasturiano/react-force-graph/blob/master/example/highlight/index.html